ipi/utils/instools.py

```diff
@@ -3,40 +3,80 @@
 from ipi.utils.messages import verbosity, info
 from ipi.utils import units
 import ipi.utils.mathtools as mt
-import os.path, glob
+import os.path, glob,copy
 
-
-def banded_hessian(h, im, shift=0.001):
+def banded_hessian(h, im, masses=True,shift=0.001):
     """Given Hessian in the reduced format (h), construct
-    the upper band hessian including the RP terms"""
+    the upper band hessian including the RP terms.
+    If masses is True returns hessian otherwise it returns dynmat"""
     nbeads = im.dbeads.nbeads
     natoms = im.dbeads.natoms
+    coef   = im.coef  #new_disc
```

Comment:
could you write in the code the reference where you took this new discretization from? does not hurt to add to the description of this function what it does, where it is from, and what it means.

Reply:
yes, certainly it helps